src/charm.py

```diff
@@ -37,21 +44,35 @@ class JenkinsAgentCharmStoredState(StoredState):
 
 
 class JenkinsAgentEnvConfig(typing.TypedDict):
-    """The environment configuration for the jenkins agent."""
+    """The environment configuration for the jenkins agent.
+
+    Attrs:
+        JENKINS_AGENTS: List of Jenkins agents.
+        JENKINS_TOKENS: List of Jenkins tokens.
+        JENKINS_URL: Jenkins server URL.
+    """
 
     JENKINS_AGENTS: str
     JENKINS_TOKENS: str
     JENKINS_URL: str
 
 
 class JenkinsAgentCharm(CharmBase):
-    """Charm for Jenkins Agent on kubernetes."""
+    """Charm for Jenkins Agent on kubernetes.
+
+    Attrs:
+        service_name: Jenkins service name.
+    """
 
     _stored = JenkinsAgentCharmStoredState()
     service_name = "jenkins-agent"
 
     def __init__(self, *args) -> None:
-        """Constructor."""
+        """Constructor.
+
+        Args:
+            args: Arguments.
+        """
```

Comment:
Trivial suggestion in case anyone is wondering what the args are haha 😄 
```suggestion
"""Initialize the instance.

        Args:
            args: arguments passed into Charmbase superclass.
        """
```

Reply:
A bit outdated now. I'm resolving